### PR TITLE
Add section about what `futhark bench` measures

### PR DESCRIPTION
**Pull request by Munksgaard**
This is an attempt at addressing my questions in #827. 

The answers have been inferred by inspecting this patch to the rodinia benchmarks created by @athas: https://github.com/diku-dk/futhark-pldi17/blob/master/rodinia_3.1-some-instrumentation.patch. By inspecting the changes to nearestNeighbor.cpp, it looks like writing to buffers is not included, but reading from buffers is.

### CODE REVIEW

File: docs/man/futhark-bench.rst [hunk 1]
```diff
@@ -104,6 +104,15 @@ OPTIONS
   ``foo.fut`` will be passed the tuning file ``foo.fut.tuning`` if it
   exists.
 
+WHAT FUTHARK BENCH MEASURES
+===========================
+
+``futhark bench`` measures the time it takes to run the given futhark
```
Comment (athas) [inline docs/man/futhark-bench.rst]: Capitalise *Futhark*.

**Comment by athas** [comment]
This is already sort of documented [here](https://futhark.readthedocs.io/en/latest/usage.html#executable-options), but in an oblique form.  I agree it is better to have it in the `futhark-bench` manpage.

**Comment by athas** [comment]
Actually, maybe the `futhark-bench` manpage should mention that it will use the `-t` option to perform the timing, and then we should actually document what `-t` means for each of the compilers.  (It means what you wrote here, so we can keep that, too.)

**Comment by Munksgaard** [comment]
> This is already sort of documented [here](https://futhark.readthedocs.io/en/latest/usage.html#executable-options), but in an oblique form. I agree it is better to have it in the `futhark-bench` manpage.

The page you link says that "writing the result" is not included in the timing, oes that mean that the final calls to `clEnqueueReadBuffer` is not included? My text says that it is.

**Comment by athas** [comment]
It is not.

**Comment by Munksgaard** [comment]
In de1e50b I've tried to move some of the explanation to the futhark-opencl man page. That way we can be more specific about how the particular backend handles timing. What do you think?

Also, how do I generate the man pages locally? It doesn't seem like they're getting installed when I run `stack build && stack install`.

**Comment by Munksgaard** [comment]
Hm, or perhaps it should just be added to https://futhark.readthedocs.io/en/latest/usage.html#opencl-specific-options ? What would you prefer?

**Comment by athas** [comment]
You generate manpages by going into `docs/` and running `make html` (you'll need Sphinx and I think something called sphinxcontrib-bibtex).

I prefer the manpages to have a quick summary of all options taken by the generated binaries, and this is a step on that path.  Your approach is better than just stuffing it in the User's Guide.

**Pull request merged**